src/Flex.php

```diff
@@ -39,13 +37,41 @@
  */
 class Flex implements PluginInterface, EventSubscriberInterface
 {
+    /**
+     * @var Composer
+     */
     private $composer;
+
+    /**
+     * @var IOInterface
+     */
     private $io;
+
+    /**
+     * @var Options
+     */
     private $options;
+
+    /**
+     * @var Configurator
+     */
     private $configurator;
+
+    /**
+     * @var Downloader
+     */
     private $downloader;
+
+    /**
+     * @var array
+     */
     private $postInstallOutput = [''];
+
+    /**
+     * @var \Closure
+     */
```

Comment:
Actually, IDEs recognize constructor arguments, but not properties that are defined in other methods, so to have autocomplete & property type recognition in IDEs we need these doc blocks because they're not initialized in constructor.

But if you really feel they're not useful, I can remove them